.github/workflows/sarif-utils-release.yml

```
@@ -45,6 +45,19 @@ jobs:
             --build-cache
             -PgprUser=${{ github.actor }}
             -PgprKey=${{ secrets.GITHUB_TOKEN }}
+      - name: gradle snapshot from branch
+        # if workflow is triggered after push to a branch, deploy snapshot
+        if: ${{ startsWith(github.ref, 'refs/heads/') }}
```

Either

```suggestion
        if: ${{ github.event_name == 'push' && github.ref_type == 'branch' && github.ref == 'refs/heads/master' }}
```

or

```suggestion
        if: ${{ github.event_name == 'push' && github.ref_type == 'branch' }}
```

&mdash; since you've already added the branch filter at the top of your YAML.

See https://github.com/0x6675636b796f75676974687562/actions-example/actions for more details.

---

Thanks for examples!

But `event_name` and `github.ref_type` are already filtered by
```bash
on:
  push:
    tags:
      - 'v*'
    branches:
      - master
```

Thus, the only difference could be in `refs/...`:
it could be `refs/tags/v123` or `refs/heads/`

So, existing `startsWith` should be enough, is not it?

---

> So, existing `startsWith` should be enough, is not it?

It will be, but it is redundant.

Basically, you're trying to differentiate between a _push of a tag_ and a _push to a named branch_. To accomplish this,

```suggestion
        if: ${{ github.ref_type == 'branch' }}
```

is enough (unless you enable the PR trigger in the future).

---

change conditions to your suggestion